crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -502,18 +515,27 @@ impl IdentityManager {
     async fn handle_new_identity(
         &self,
         response: &KeysQueryResponse,
+        maybe_verified_own_identity: Option<&OwnUserIdentity>,
         master_key: MasterPubkey,
         self_signing: SelfSigningPubkey,
         changed_private_identity: &mut Option<PrivateCrossSigningIdentity>,
     ) -> Result<UserIdentityData, SignatureError> {
         if master_key.user_id() == self.user_id() {
+            // Own identity
             let user_signing = self.get_user_signing_key_from_response(response)?;
             let identity = OwnUserIdentityData::new(master_key, self_signing, user_signing)?;
             *changed_private_identity = self.check_private_identity(&identity).await;
             Ok(identity.into())
         } else {
             // First time seen, create the identity. The current MSK will be pinned.
             let identity = OtherUserIdentityData::new(master_key, self_signing)?;
+            let is_verified = maybe_verified_own_identity.map_or(false, |own_user_identity| {
+                own_user_identity.is_identity_signed(&identity).is_ok()
+            });
+            if is_verified {
+                identity.mark_as_verified_once();
```

Comment:
Wait, do the tests in the previous commit fail without this? I would like those tests to be in the same commit that makes them pass, or a later one.

Reply:
Now I see why you called the commit "TDD" - because the tests were written before the code! That is awesome during development, but we don't want commits in the history that fail tests because it is much harder to `git bisect` to find what caused a problem.

Reply:
So now, as I can't force push, I have to wait the end of the review, then reset all to re-order the changes and then force push?
Won't all of this squashed at the end?

Reply:
No, we don't squash when we merge: after review, we force push a sensible set of commits, then merge with "Rebase and merge".